internal/pkg/deploy/cloudformation/stack/backend_svc.go

```diff
@@ -219,6 +237,11 @@ func (s *BackendService) httpLoadBalancerTarget() (targetContainer *string, targ
 		targetPort = s.manifest.Sidecars[aws.StringValue(targetContainer)].Port
 	}
 
+	// Route load balancer traffic to the target_port if mentioned.
+	if s.manifest.RoutingRule.TargetPort != nil {
+		targetPort = s.manifest.RoutingRule.TargetPort
+	}
```

Comment:
Do we not need to figure out the `targetContainer` if a user specified a `targetPort`? 

My understanding is that we still need to provide the appropriate ContainerName in the `loadBalancers` setting of the service so we would need a way of looking the container name of the port.

If I have the following manifest for example:
```
image:
  build: './Dockerfile'
  port: 8080 
  
http:
  path: '/'
  target_port: 80

sidecars:
  nginx:
    port: 80
```
In this situation, I would expect the ContainerName to be set to nginx rather than the main container because my `target_port` matches the exposed port in the manifest.

Reply:
Yes you are correct. I am making that change so that target container is set to the right container name. 
Also just wants to make sure my understanding aligns with yours. There could be following scenarios --
1. if `target_container` is set then target container's port is target port. 
2. if `target_port` is set then we will check if `target_container` is set? if yes, then we know target container and target port. If `target_container` is not set then we will check if sidecar container port is same as that of `target_port` and if found then set that sidecar container name as target container name. Otherwise we will set main container name as target container name.

Reply:
Your summary sounds good to me :shipit: